Sources/FoundationEssentials/Platform.swift

```diff
@@ -174,7 +174,7 @@ extension Platform {
         // FIXME: bionic implements this as `return 0;` and does not expose the
         // function via headers. We should be able to shim this and use the call
         // if it is available.
-#if !os(Android)
+#if !os(Android) && !os(WASI)
         guard issetugid() == 0 else { return nil }
```

Comment:
Just confirming: since WASI doesn't have the concept of users there's no equivalent that we should be calling instead, right?

Reply:
Right, WASI doesn't expose "taintedness" concept nor secure env var